### PR TITLE
feat(medusa-plugin-sendgrid): Add error messages

### PR DESCRIPTION
**Pull request by pevey**
This PR adds a few error messages in the sendNotification method of the SendgridService to assist with troubleshooting and configuration.  Several users have reported "sendgrid is not sending notifications."  This PR will help those users provide more useful information when reporting issues.

### CODE REVIEW

File: packages/medusa-plugin-sendgrid/src/services/sendgrid.js [hunk 1]
```diff
@@ -221,10 +222,14 @@ class SendGridService extends NotificationService {
     let templateId = this.getTemplateId(event)
 
     if (!templateId) {
-      return false
+      throw new MedusaError(MedusaError.Types.INVALID_DATA, "Sendgrid service: No template was set for this event")
```
Comment (olivermrbl) [inline packages/medusa-plugin-sendgrid/src/services/sendgrid.js]: thought(non-blocking): Should we also log the event name?

Comment (pevey) [reply packages/medusa-plugin-sendgrid/src/services/sendgrid.js]: Does that string get interpolated?  Can I do:

```
`Sendgrid service: No template was set for event: ${event}`
```

Comment (adrien2p) [reply packages/medusa-plugin-sendgrid/src/services/sendgrid.js]: yes you can do that 👍

**Review by olivermrbl — APPROVED** [review]
lgtm!

**Review by adrien2p — APPROVED** [review]
LGTM

**Pull request merged**